### ISSUE TITLE
Addition of "num_cycles" parameter to the cosine_with_restarts scheduler in train_unconditional.py

### ISSUE THREAD

**Issue by lukewardford** [huggingface/diffusers#3195]
In the [train_unconditional.py ](https://github.com/huggingface/diffusers/blob/main/examples/unconditional_image_generation/train_unconditional.py) script, I noticed that even after choosing the cosine_with_restarts learning rate scheduler. it's acts just like a normal cosine scheduler. That's because the [get_scheduler](https://github.com/huggingface/diffusers/blob/main/src/diffusers/optimization.py#L238) method have a default number of cycles which is 1. I can't change that values without modifying the script. I'm suggesting the addition of a new command line parameter called "num_cycles" which would determine the number of cycles the scheduler should perform.

**Comment by patrickvonplaten** [comment]
Would be ok for me to add a `num_cycles` argument, but agree that it's not super important

**Comment by sneccc** [comment]
> Would be ok for me to add a `num_cycles` argument, but agree that it's not super important

@patrickvonplaten if possible the addition of  minimum learning rate for cosine with restarts based on [https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.CosineAnnealingWarmRestarts.html#torch.optim.lr_scheduler.CosineAnnealingWarmRestarts](url) so it doesnt reach 0 lr which is not optimal in training
